src/common/control.rs

```diff
@@ -28,147 +25,135 @@ pub struct Control {
 
 
 impl Control {
-pub fn init() -> Control {
-    // create a Control instance
-    let mut ctl = Control {
-        state: false,
-        pins: MyPins::new(),
-    };
-
-    // add the GPIO pins to it
-    ctl.pins.add_pin(CW, Direction::Low);
-    ctl.pins.add_pin(CCW, Direction::Low);
-    ctl.pins.add_pin(ESTOP, Direction::In);
-
-    // return to caller
-    ctl
-}
+    pub fn init() -> Control {
+        // create a Control instance
+        let mut ctl = Control {
+            state: false,
+            pins: MyPins::new(),
+        };
+
+        // add the GPIO pins to it
+        ctl.pins.add_pin(CW, Direction::Low);
+        ctl.pins.add_pin(CCW, Direction::Low);
+        ctl.pins.add_pin(ESTOP, Direction::In);
+
+        // return to caller
+        ctl
+    }
 
 
-///////////////////////////////////////////////////////////////////////////////
-// Function Name: update
-//
-// Purpose: Reads sensor data from shared memory, executes the control
-//          algorithm, and updates GPIO pin states in shared memory.
-//
-// INPUTS: mem -- reference to shared memory
-//
-// RETURNS: Ok(0)    -- all is well; continue running
-//          Ok(1)    -- shut down
-//          Err(err) -- an error occurred; shut down
-//
-///////////////////////////////////////////////////////////////////////////////
-pub fn update(&mut self, mem: &mut SharedMemory) -> Result<u8, String> {
-    // stop_pin <- Get the value of the ESTOP pin
-    let stop_pin = try!(self.pins.get_value(ESTOP));
-
-    // IF stop_pin is 1
-    if false && stop_pin == SHUT_DOWN {
-        // RETURN 1
-        return Ok(stop_pin);
-    } // END IF
-
-    // rate_x <- READ the gyro's x axis rate from shared memory
-    let rate_x = mem.gyro_y - 0.41; // Offsetting the led edison board
-	println!("gyro rate {}", rate_x);
-
-    const ACTIVATION_THRESHOLD: f32 = 0.175;
-    // IF rate_x GE 0.175
-    if rate_x >= ACTIVATION_THRESHOLD {
-        // CALL state_update(rate_x)
-        // CALL write_pin(CW_pin, new state value, reference to shared memory)
-        self.state_update(rate_x);
-        let value = self.state as u8;
-        try!(self.write_pin(CW, value, mem));
-    } // ELSE IF rate_x LE -0.175
-    else if rate_x <= -ACTIVATION_THRESHOLD {
-        // CALL state_update(rate_x)
-        // CALL write_pin(CCW_pin, new state value, reference to shared memory)
-        self.state_update(rate_x);
-        let value = self.state as u8;
-        try!(self.write_pin(CCW, value, mem));
-    } // ELSE
-    else {
-        // turn off both gpio pins and update their state
-        // CALL write_pin(CW_pin, 0, mem)
-        // CALL write_pin(CCW_pin, 0, mem)
-        try!(self.write_pin(CW, 0, mem));
-        try!(self.write_pin(CCW, 0, mem));
-    } // END IF
-
```

Comment:
I think that rustdoc comments work on libraries but not binaries. So, the common files can have // style comments, but test_mode/src and filght_mode/src should have /// style comments.